misc/kaprekars

```diff
@@ -0,0 +1,105 @@
+/*
+Ioannis Sykas Leriotis 29/09/2019
+6174 is the Kaprekar Constant. This number is special as we always get this number when following steps are followed 
+for any four digit number such that all digits of number are not same.
+
+Sort four digits in ascending order and store result in a number “asc”.
+Sort four digits in descending order and store result in a number “desc”.
+Subtract number larger number from smaller number, i.e., abs(asc – desc).
+Repeat above three steps until the result of subtraction doesn’t become equal to the previous number.
+*/
+
+# include <stdio.h>
+# include <stdbool.h>
+# include <math.h>
+
+int kaprekarRec(int n, int prev)
+{
+    int i,j,a,count=0;
+    if (n == 0){
+       return 0;
+    }
+    // Store current n as previous number
+    prev = n;
+
+    // Get four digits of given number
+    int digits[4];
+    for (i=0; i<4; i++)
+    {
+       digits[i] = n%10;
+       n = n/10;
+       count++;
+    }
+     // Get all four dgits in descending order
+    // in the form of number "desc"
+      for (i = 0; i < count; i++)
+        {
+            for (j = i + 1; j < count; j++)
+            {
+                if (digits[i] < digits[j])
+                {
+                    a = digits[i];
+                    digits[i] = digits[j];
+                    digits[j] = a;
+                }
+            }
+
+        }
+    int desc = 0;
+    for (i=0; i<4; i++){
+       desc = desc*10 + digits[i];
+    }
+
+    // Sort all four dgits in ascending order
```

Comment:
you have already sorted the digits in descending order in ``digits`` array. Instead of again sorting the array in ascending order, you could just iterate through the array in reverse to get the ``asc``